### ISSUE TITLE
[Bug]: error trying to use grafana_folder_permission_item (Resource)

### ISSUE THREAD

**Issue by AkakievKD** [grafana/terraform-provider-grafana#1485]
### Terraform Version

v1.7.5

### Terraform Grafana Provider Version

2.17

### Grafana Version

10.4.0

### Affected Resource(s)

grafana_folder_permission_item

### Terraform Configuration Files

```
i use module

resource "grafana_folder_permission_item" "permission_item" {
  depends_on = [grafana_folder.kubernetes_folder]
  for_each   = { for key, value in var.grafana_folder_permission_item : key => value }
  folder_uid = each.key
  team       = each.value.team
  permission = each.value.permission
  role       = each.value.role
  user       = each.value.user
}

variable "grafana_folder_permission_item" {
  type = map(object({
    team       = optional(string, null)
    permission = string
    user       = optional(string, null)
    role       = optional(string, null)
  }))
  default = {}
}

module "grafana" {
  source = "../modules/grafana-oss-module"
  
  grafana_folder_permission_item = {
    "kuber" = {
      permission = "Admin"
      user       = "11"
    }
  }
}

terraform {
  required_providers {
    grafana = {
      source = "grafana/grafana"
    }
  }
}

provider "grafana" {
  url  = "https://grafana.cloud-infra.ru/"
  auth = "admin:passwd"
}
```

### Expected Behavior

terraform works with resource without errors

because with other resources terraform works correct

### Actual Behavior

<img width="1284" alt="Screenshot at Apr 13 15-37-03" src="https://github.com/grafana/terraform-provider-grafana/assets/82488489/7a1b34e6-1311-4a76-b02f-cf391fb22922">

### Steps to Reproduce

standard steps

### Important Factoids

grafana in kubernetes
<img width="485" alt="Screenshot at Apr 13 15-41-26" src="https://github.com/grafana/terraform-provider-grafana/assets/82488489/652a9935-fdde-4d81-8ecf-4a61e0c7dc39">

**Comment by julienduchesne** [comment]
🤔 You're right. No idea why the tests are passing for those. We are currently implementing new resources with the latest Terraform provider plugin framework (internal SDK) and it seems like there's an issue with the client validation there. I will fix this ASAP!

**Comment by bcrisp4** [comment]
Thanks for fixing! Any idea when the next release will be? I'd really like to use the new grafana_folder_permission_item resource

**Comment by julienduchesne** [comment]
Before EOW

**Comment by whiteninja76** [comment]
@julienduchesne Did the fix for this get released in 2.18.0 ?
